lib/std/fs.zig

```diff
@@ -369,6 +370,10 @@ pub const Dir = struct {
             fn nextSolaris(self: *Self) !?Entry {
                 start_over: while (true) {
                     if (self.index >= self.end_index) {
+                        if (self.first) {
```

Comment:
Aren't these conditions backwards? We only want to `lseek_SET` when it's *not* the first iteration, right?

Reply:
No. We should rewind the directory before reading entries, and only on the first iteration. When else would we do it?

Reply:
Ah, you're right, sorry, had things mixed up in my head.